logictest: do not run ARM builds in mixed version tests with versions less than v22.2

pkg/sql/logictest/tests/cockroach-go-testserver-22.1-22.2/cockroach-go-testserver-22_1-22_2_test.TestLogic_testserver_test_22_1_22_2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8278267?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8278267?buildTab=artifacts#/) on master @ [dc273e05120f28f9ef9a8f8eb60ef678fc105756](https://github.com/cockroachdb/cockroach/commits/dc273e05120f28f9ef9a8f8eb60ef678fc105756):


```
=== RUN   TestLogic_testserver_test_22_1_22_2
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f0512f3559a6f1fc0be49477689a2598/logTestLogic_testserver_test_22_1_22_2994206753
    test_log_scope.go:79: use -show-logs to present logs inline
    logic.go:1838: 
        :0: error while processing
    logic.go:1838: unexpected HTTP response from https://binaries.cockroachdb.com/cockroach-v22.1.6.linux-arm64.tgz?ci=true: 404
        <html><body>
        <h1>404 Error: Not Found</h1>
        </body></html>
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f0512f3559a6f1fc0be49477689a2598/logTestLogic_testserver_test_22_1_22_2994206753
--- FAIL: TestLogic_testserver_test_22_1_22_2 (0.52s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_testserver_test_22_1_22_2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-23302

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1819,6 +1819,9 @@ func (t *logicTest) setup(
 
 	if cfg.UseCockroachGoTestserver {
 		skip.UnderStress(t.t(), "test takes a long time and downloads release artifacts")
+		if runtime.GOARCH == "arm64" && strings.HasPrefix(cfg.CockroachGoBootstrapVersion, "v22.1") {
+			skip.IgnoreLint(t.t(), "Skip under ARM64. There are no ARM release artifacts for v22.1.")
+		}
 		if !bazel.BuiltWithBazel() {
 			skip.IgnoreLint(t.t(), "cockroach-go/testserver can only be uzed in bazel builds")
 		}

